### ISSUE TITLE
Instructions for settings up HIE with hoogle DB

### ISSUE THREAD

**Issue by teto** [domenkozar/hie-nix#27]
It might be obvious to haskell pros but as a beginner, I had to experiment a bit before getting right.

I've installed hie-nix globally (via nix-env), then ran `nix-shell -p 'haskellPackages.hoogleWithPackages(p: with p; [netlink])' `. I use hie in neovim as an LSP server.
It kept failing on importing modules/finding hoogle db.

Looking at ghc wrapper and exporting these variables solved the problem for me
```
export NIX_GHC_LIBDIR='/nix/store/cdnv4gdimys29pk0s2g38f78ibggc84y-ghc-8.4.3-with-packages/lib/ghc-8.4.3'
export NIX_GHC_DOCDIR='/nix/store/cdnv4gdimys29pk0s2g38f78ibggc84y-ghc-8.4.3-with-packages/share/doc/ghc/html'
```

Feel free to close, even if it's not in readme, the issue itself might help others.

**Comment by domenkozar** [comment]
Interesting, it seems like those paths are what GHC in Nix expects to find things like hoogle DB. Not sure how to fix that best yet, but good find!

**Comment by domenkozar** [comment]
@expipiplus1 is there a snippet we could add to FAQ?

**Comment by teto** [comment]
After a talk on IRC, I ended up even more confused. I guess my question is how to setup everything so that hie can pickup the nix-shell's ghc ? I don't get how it's possible right now and consequently make hie useless since it doesn't find any dependencies. If someone could share their worfklow/confirm they have a functional hie, that would be a big help

**Comment by teto** [comment]
Kudos to alanz and benkolera for helping me find an issue to these problems.
1/ the hie --vomit flag goes to stderr, LanguageClient_neovim can save stderr to a specific file. 
2/ Looking at the stderr showed hie was looking  for information in "dist" subfolders. Seems like there are different types of project and  I use exclusively nix, not cabal so that was not good. Removing "dist" should have been enough but for some reason the folder gets recreated everytime I start neovim (not sure if it's hie's fault or one of my plugins). Removing the .cabal file solves it.

On a sidenote, hie now accepts an environment variable HIE_HOOGLE_DATABASE so I added a shellHook to export it otherwise hie couldn't find my hoogle database. I built the path through NIX_GHC_LIBDIR since NIX_GHC_DOCDIR doesn't seem to exist for my project :s 

To summarize what looks like what I believe the easiest way to use hie with nix:
```
haskellPackages.shellFor {
  packages = p: with p; [ haskellPackages.netlink-pm ];
  withHoogle = true;
  nativeBuildInputs = [ hie ];

  # export HIE_HOOGLE_DATABASE=$NIX_GHC_DOCDIR as DOCDIR doesn't exist it won't work
  shellHook = ''
    export HIE_HOOGLE_DATABASE="$NIX_GHC_LIBDIR/../../share/doc/hoogle/index.html"
  '';
}
```

**Comment by expipiplus1** [comment]
I extract the database location from the hoogle wrapper: https://github.com/expipiplus1/vector-sized/pull/74/files#diff-5712e736e0de6ba170577f8472c398e9R31

**Comment by expipiplus1** [comment]
I've opened a PR for the FAQ, @domenkozar . Sorry for the wait.

**Comment by domenkozar** [comment]
Thank you <3
